src/classes/ADDR_Addresses_TEST.cls

```diff
@@ -998,6 +998,254 @@ public with sharing class ADDR_Addresses_TEST {
 
     /*********************************************************************************************************
     @description
+        Update Contact's mailing address (with State & Country Picklists enabled) and mark it as override,
+        to cause creation of a new non-default Address record with the State & Country values set as expected.
+    verify:
+        Contact gets new address
+        HH sticks with default Address
+        New Address created with State value set as expected
+    **********************************************************************************************************/
+    static testMethod void testContactAddressOverrideForStateChangeWithStateAndCountryPicklistEnabled() {
+        if (!ADDR_Addresses_TDTM.isStateCountryPicklistsEnabled) {
+            return;
+        }
+
+        if (strTestOnly != '*' && strTestOnly != 'testContactAddressMarkOverrideWithStateAndCountryPicklistsEnabled') {
+            return;
+        }
+
+        Integer hhSize = 1;
+        Integer contactSize = 1;
+
+        // This creates a default Address for each HH
+        createHHTestData(hhSize, contactSize);
+        System.assertEquals(1, getAddresses().size());
+
+        // Modify the Contact addresses directly, and set override
+        Contact con = listConT[0];
+        con.put('MailingStateCode', 'CA');
+        con.put('MailingCountryCode', 'US');
+        con.is_Address_Override__c = true;
+
+        Test.startTest();
+        update con;
+        Test.stopTest();
+
+        // Verify a new Address record was created
+        List<Address__c> listAddr = getAddresses();
+        System.assertEquals(2, listAddr.size());
+
+        // Populate a Map of AccountId to Address record
+        Map<Id, Address__c> accountIdToAddress = new Map<Id, Address__c>();
+        for(Address__c eachAddress : listAddr) {
+            if(eachAddress.Default_Address__c) {
+              accountIdToAddress.put(eachAddress.Household_Account__c, eachAddress);
+            }
+        }
+
+        // Verify the Account's Billing Address info matches the Default Address record
+        List<Account> retrievedAccounts = Database.query(buildAccountSoql());
+        assertEquals(listAddrT[0], accountIdToAddress.get(retrievedAccounts[0].Id));
+
+        // Verify the Contact's Address was updated as expected and that the Household and Contact Addresses are different.
+        Map<Id, Account> accountIdToAccount = new Map<Id, Account>(retrievedAccounts);
+        List<Contact> listCon = Database.query(buildContactSoql());
+        for (Contact eachContact : listCon) {
```

Comment:
It seems we have one Account and one Contact in the test. Do we need to use the `for` loop? It is the good approach if we have another Contact for which Address has not changed, but since we don't, we might extract the Contact into a variable or use `listCon[0]` instead of the `for` loop?